newlib/libc/locale/newlocale.c

```diff
@@ -113,12 +113,15 @@ newlocale (int category_mask, const char *locale,
   (void) locale;
   (void) base;
 
+#ifdef _MB_CAPABLE
```

Comment:
These look like pretty simple parameter validation checks; is there some reason they aren't valid when `_MB_CAPABLE` isn't defined?

Reply:
Thank you for pointing this out, Keith. You’re absolutely right -- these parameter validation checks are general and not tied to `_MB_CAPABLE`. I added the guard mistakenly, thinking it might be relevant to multibyte-specific logic, but I now realize that was an incorrect assumption.

I’ll go ahead and remove the guard so that these checks apply universally, as they should. Thanks for catching this!